scripts/test/hyriseServer_test.py

```diff
@@ -9,27 +10,31 @@
 def main():
     build_dir = initialize()
 
-    arguments = {}
-    arguments["--benchmark_data"] = "tpc-h:0.01"
+    server = pexpect.spawn(f"{build_dir}/hyriseServer --benchmark_data=tpc-h:0.01 -p 0", timeout=10)
 
-    benchmark = pexpect.spawn(f"{build_dir}/hyriseServer --benchmark_data=tpc-h:0.01 -p 0", timeout=10)
-
-    benchmark.expect_exact("Loading/Generating tables", timeout=120)
-    benchmark.expect_exact("Encoding 'lineitem'", timeout=120)
+    server.expect_exact("Loading/Generating tables", timeout=120)
+    server.expect_exact("Encoding 'lineitem'", timeout=120)
     search_regex = r"Server started at 0.0.0.0 and port (\d+)"
-    benchmark.expect(search_regex, timeout=120)
-
-    server_port = int(re.search(search_regex, str(benchmark.after)).group(1))
-    client = pexpect.spawn(f"psql -h localhost -p {server_port}", timeout=20)
+    server.expect(search_regex, timeout=120)
+
+    server_port = int(re.search(search_regex, str(server.after)).group(1))
+    # Recent Postgres/psql versions changed the authentication behavior, resulting in connection errors on some setups.
+    # Disabling encrypted connections solves the issue. Since hyriseServer does not implement authentication at all,
+    # this is no problem.
+    # See https://github.com/psycopg/psycopg2/issues/1084#issuecomment-656778107 and
+    # https://www.postgresql.org/docs/13/libpq-connect.html#LIBPQ-CONNECT-GSSENCMODE
+    environment_variables = os.environ.copy()
+    environment_variables.update({"PGGSSENCMODE": "disable"})
```

Comment:
Comment only: not using
```python
environment_variables = os.environ | {"PGGSSENCMODE": "disable"}
```
as it is Python 3.9+ syntax and we require Python 3.6+.

Reply:
Just a side note: if you see any reason to update to 3.9, feel free. Python 3.6 is six years old and reached its end of life.

Reply:
Could also be
```python
environment_variables["PGGSSENCMODE"] = "disable"
```
so guess it's just a matter of taste. However, updating some of our (python) dependencies could be done, right. But if we do so, we should maybe also refactor our scripts with new python functionality and more pythonic approaches (code in main function rather than globally, argparse, etc) in general.

Reply:
I doubt somebody finds the time for such "pythonications" if we cannot automate them.

Reply:
Don't think so either. Thus, I guess updating Python is not necessary now.